src/DebugProxy.ts

```diff
@@ -0,0 +1,183 @@
+                        });
+                        socket.resume();
+                    });
+
+                    this._wsclient.on('connectFailed', (err: Error) => {
+                        this._outputChannel.appendLine(`[WebSocket] ${err}`);
+                        this.dispose();
+                        socket.destroy();
+                        this.emit('error', err);
+                    });
+
+                    this._wsclient.connect(
+                        `wss://${this._siteWrapper.appName}.scm.azurewebsites.net/DebugSiteExtension/JavaDebugSiteExtension.ashx`,
+                        undefined,
+                        undefined,
+                        { 'Cache-Control': 'no-cache', Pragma: 'no-cache' },
+                        { auth: `${this._publishProfile.publishingUserName}:${this._publishProfile.publishingPassword}` }
+                    );
+                }
+
+                socket.on('data', (data: Buffer) => {
+                    if (this._wsconnection) {
+                        this._wsconnection.send(data);
+                    }
+                });
+
+                socket.on('end', () => {
+                    this._outputChannel.appendLine(`[Proxy Server] client disconnected ${socket.remoteAddress}:${socket.remotePort}`);
+                    this.dispose();
+                    this.emit('end');
+                });
+
+                socket.on('error', (err: Error) => {
+                    this._outputChannel.appendLine(`[Proxy Server] ${err}`);
+                    this.dispose();
+                    socket.destroy();
+                    this.emit('error', err);
+                });
+            });
+
+            this._server.on('listening', () => {
+                this._outputChannel.appendLine('[Proxy Server] start listening');
+                this.emit('start');
+            });
+
+            this._server.listen({
+                host: 'localhost',
+                port: this._port,
+                backlog: 1
+            });
+        }
+    }
+
+    public dispose(): void {
+        if (this._wsconnection) {
+            this._wsconnection.close();
+            this._wsconnection = undefined;
+        }
+        this._wsclient = undefined;
+        if (this._server) {
+            this._server.close();
+            this._server = undefined;
+        }
+        this._keepAlive = false;
+    }
+
+    //keep querying the function app state, otherwise the connection will lose.
+    private async keepAlive(): Promise<void> {
+        if (this._keepAlive) {
+            try {
+                await this.getFunctionState();
+                setTimeout(this.keepAlive, 60 * 1000 /* 60 seconds */);
+            } catch (err) {
+                this._outputChannel.appendLine(`[Proxy Server] ${err}`);
+                setTimeout(this.keepAlive, 5 * 1000 /* 5 seconds */);
+            }
+        }
+    }
+
+    private async getFunctionState(): Promise<void> {
```

Comment:
Great, using the function in npm package will be much better, could you please help expose that function? Thanks. 😄

Reply:
Okay it's exposed in v0.10.0 of the vscode-azureappservice npm package

Reply:
Thank you! 😄 

I update the `vscode-azureappservice` to 0.10.0, meanwhile the `vscode-azureextensionui` is updated to 0.7.1 because of the version updated for `vscode-azureappservice`.

After version update, some files have to be changed because of the change of API.

Reply:
@EricJizbaMSFT Thank you 😄 

FYI, `src/commands/configureDeploymentSource.ts` This file is changed according to the API change.